packages/server/src/integrations/googlesheets.ts

```diff
@@ -434,7 +434,19 @@ class GoogleSheetsIntegration implements DatasourcePlus {
     try {
       await this.connect()
       const sheet = this.client.sheetsByTitle[query.sheet]
-      const rows = await sheet.getRows()
+      let rows: GoogleSpreadsheetRow[] = []
+      if (query.paginate) {
+        let page: number =
+          typeof query.paginate.page === "number"
+            ? query.paginate.page
+            : parseInt(query.paginate.page || "1")
+        rows = await sheet.getRows({
+          limit: query.paginate.limit,
```

Comment:
Should we fall back to some sensible value for limit in case it isn't specified? Something like 50?

Reply:
Good call. I went for 100 as this matches the pagination for the Internal DB